controlplane/kubeadm/api/v1alpha3/kubeadm_control_plane_webhook.go

```diff
@@ -229,6 +231,30 @@ func (in *KubeadmControlPlane) validateCommon() (allErrs field.ErrorList) {
 	return allErrs
 }
 
+func (in *KubeadmControlPlane) validateEtcd() (allErrs field.ErrorList) {
```

Comment:
This will need to accept `prev` as an argument to do a check between previous and new

Reply:
actually, what exists should be sufficient, just the test would need updating

Reply:
ok sorry for the back and forth haha, i'm thinking about various use cases. 

But I've got an example that this should cover:

If an existing config defines local
then the user manually removes the local config and supplies an external config

the check that only looks at `in`'s configuration is not sufficient and will need to see if previous had either local or external and ensure that the incoming structure does not define the opposite while removing the old one.

Does that make sense? That felt hard to explain so I'd be happy to hop on a call to explain further if necessary

Reply:
@chuckha please take a look